src/drt/src/dr/FlexDR.cpp

```diff
@@ -2328,6 +2328,148 @@ static std::vector<FlexDR::SearchRepairArgs> strategy()
+      gtl::polygon_90_data<frCoord> poly;
+      vector<gtl::point_data<frCoord>> points;
+      for (auto point : rect.getPoints()) {
+        points.push_back({point.x(), point.y()});
+      }
+      poly.set(points.begin(), points.end());
+      routeSetByLayerNum[pwire->getLayerNum()] += poly;
+    }
+    for (const auto& via : net->getVias()) {
+      {
+        odb::Rect rect;
+        via->getLayer1BBox(rect);
+        gtl::polygon_90_data<frCoord> poly;
+        vector<gtl::point_data<frCoord>> points;
+        for (auto point : rect.getPoints()) {
+          points.push_back({point.x(), point.y()});
+        }
+        poly.set(points.begin(), points.end());
+        routeSetByLayerNum[via->getViaDef()->getLayer1Num()] += poly;
+      }
+      {
+        odb::Rect rect;
+        via->getLayer2BBox(rect);
+        gtl::polygon_90_data<frCoord> poly;
+        vector<gtl::point_data<frCoord>> points;
+        for (auto point : rect.getPoints()) {
+          points.push_back({point.x(), point.y()});
+        }
+        poly.set(points.begin(), points.end());
+        routeSetByLayerNum[via->getViaDef()->getLayer2Num()] += poly;
+      }
+    }
+
+    for (const auto& shape : net->getOrigGuides()) {
+      odb::Rect rect;
+      shape.getBBox(rect);
+      gtl::polygon_90_data<frCoord> poly;
+      vector<gtl::point_data<frCoord>> points;
+      for (auto point : rect.getPoints()) {
+        points.push_back({point.x(), point.y()});
+      }
+      poly.set(points.begin(), points.end());
+      guideSetByLayerNum[shape.getLayerNum()] += poly;
+    }
+
+    for (frLayerNum lNum = 0; lNum < getTech()->getLayers().size(); lNum++) {
+      if (getTech()->getLayer(lNum)->getType() != dbTechLayerType::ROUTING
+          || lNum > TOP_ROUTING_LAYER)
+        continue;
+      if (routeSetByLayerNum[lNum].empty()) {
+        #pragma omp critical
+        netsCoverage[net.get()].push_back(-1.0);
+        continue;
+      }
+      auto routingArea = gtl::area(routeSetByLayerNum[lNum]);
+      auto coveredArea
+          = gtl::area(routeSetByLayerNum[lNum] & guideSetByLayerNum[lNum]);
+      auto coveredPercentage = (coveredArea / (routingArea * 1.0)) * 100;
+      #pragma omp critical
+      {
+        netsCoverage[net.get()].push_back(coveredPercentage);
+        totalAreaByLayerNum[lNum] += routingArea;
+        totalCovereAreaByLayerNum[lNum] += coveredArea;
+      }
+    }
+  }
+  for (auto [net, coverage] : netsCoverage) {
+    file << net->getName() << ",";
+    for (auto coveredPercentage : coverage)
+      if (coveredPercentage < 0.0)
+        file << "NA,";
+      else
+        file << fmt::format("{:.2f}%,", coveredPercentage);
+    file << std::endl;
+  }
+  file << "Total,";
+  unsigned long long totalArea = 0;
+  unsigned long long totalCoveredArea = 0;
+  for (const auto& layer : getTech()->getLayers()) {
+    if (layer->getType() == dbTechLayerType::ROUTING
+        && layer->getLayerNum() <= TOP_ROUTING_LAYER) {
```

Comment:
It would be nice to have a getRoutingLayers() rather than repeating this filtering everywhere.